ledger/ledger.go

```diff
@@ -248,6 +257,36 @@ func (l *Ledger) GetAccount(address string) (core.Account, error) {
 	return account, nil
 }
 
-func (l *Ledger) SaveMeta(ty string, id string, m core.Metadata) error {
-	return l.store.SaveMeta(ty, id, m)
+func (l *Ledger) SaveMeta(targetType string, targetID string, m core.Metadata) error {
+	l.Lock()
+	defer l.Unlock()
+
+	if l._lastMetaID == 0 {
+		count, err := l.store.CountMeta()
+		if err != nil {
+			return err
+		}
+		l._lastMetaID = int(count) - 1
+	}
+
+	timestamp := time.Now().Format(time.RFC3339)
+
+	for key, value := range m {
+		metaRowID := fmt.Sprint(l._lastMetaID + 1)
+
+		err := l.store.SaveMeta(
+			metaRowID,
+			timestamp,
+			targetType,
+			targetID,
+			key,
+			string(value),
+		)
+		if err == nil {
```

Comment:
Perhaps return earlier here:
```go
if err != nil {
  return err
}
l._lastMetaID++
```

Reply:
good call, not sure what I was thinking here 😄